onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -286,8 +286,8 @@ def __init__(self, path_or_bytes, sess_options=None, providers=None, provider_op
         :param sess_options: session options
         :param providers: Optional sequence of providers in order of decreasing
             precedence. Values can either be provider names or tuples of
-            (provider name, options dict). If not provided, then all available
-            providers are used with the default precedence.
+            (provider name, options dict). If not provided, then 'CPUExecutionProvider'
```

Comment:
thanks for updating the doc.
what you wrote is slightly inaccurate. The only time you can leave off providers is if your build *only* supports CPUExecutionProvider.
If it can support other providers, then you must explicitly specify the providers(s) you want to use. that is what lines 359-363 enforces.

Reply:
The [lines 337-342](https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/python/onnxruntime_inference_collection.py#L337-L342) handle the error raised by line  359-363 when `fall_back` is enabled. Line 328 sets `self._enable_fallback = True` by default. The below message is what I got if I set `providers=None`, but the session is valid and the program did not fail.
```
EP Error using None
Falling back to ['CPUExecutionProvider'] and retrying.
```
How about adding one more sentence "For using other providers, please explicitly specify the providers."?

Reply:
ok that's fine. I guess that is easier to document and explain.

Reply:
Just update it. Please take another look.